src/condor_schedd.V6/schedd.cpp

```diff
@@ -1384,10 +1384,21 @@ Scheduler::count_jobs()
 	if (FlockCollectors) {
 		FlockCollectors->rewind();
 		Daemon *daemon;
+		StringList effectFlockList;
+		int currLevel = 0;
 		while (FlockCollectors->next(daemon)) {
 			auto col = static_cast<DCCollector*>(daemon);
 			FlockPools.insert(col->name());
+			if (currLevel < FlockLevel)
+				effectFlockList.append(col->addr());
+			++currLevel;
 		}
+
+		if (!effectFlockList.isEmpty()) {
+			char* flockList = effectFlockList.print_to_string();
```

Comment:
this can be 
    auto_free_ptr flockList(effectiveFlockList...)
 and then you don't need to remember to free it.